lib/ldap_fluff/netiq_member_service.rb

```diff
@@ -0,0 +1,81 @@
+require 'net/ldap'
+
+# handles the naughty bits of posix ldap
+class LdapFluff::NetIQ::MemberService < LdapFluff::GenericMemberService
+
+  def initialize(ldap, config)
+    @attr_login = (config.attr_login || 'memberuid')
+    super
+  end
+
+  def find_user(uid, base_dn = @base)
+    user = @ldap.search(:filter => name_filter(uid), :base => base_dn)
+    raise UIDNotFoundException if (user.nil? || user.empty?)
+    user
+  end
+
+  def find_by_dn(search_dn)
+    entry, base = search_dn.split(/(?<!\\),/, 2)
+    _entry_attr, entry_value = entry.split('=', 2)
+    entry_value = entry_value.gsub('\,', ',')
+    user = @ldap.search(:filter => name_filter(entry_value, "workforceid"), :base => base)
+    raise self.class::UIDNotFoundException if (user.nil? || user.empty?)
+    user
+  end
+
+  def get_logins(userlist)
+    userlist.map(&:downcase!)
+    result = []
+
+    userlist.each do |current_user|
+       result << find_by_dn(current_user)[0][@attr_login][0]
+    end
+
+    result
+  end
+
+  # return an ldap user with groups attached
+  # note : this method is not particularly fast for large ldap systems
```

Comment:
Could we incorporate the changes from https://github.com/theforeman/ldap_fluff/pull/63 to make it faster?

Reply:
Given the updates, is this still slow? Or in other words, is the note still relevant or should it be removed?

Reply:
I have not tested it, yet, but probably at least less slow as before :grin: